src/collapse/perf.rs

```diff
@@ -82,35 +92,59 @@ pub struct Folder {
     /// Function entries on the stack in this entry thus far.
     stack: VecDeque<String>,
 
-    opt: Options,
+    // Options...
```

Comment:
Why do you want to split this out into its fields rather than just keeping an `Options` like before?

Reply:
Has to do with me combining the `event_filtering: EventFilterState` field of `perf::Folder` and the `pub event_filter: Option<String>` field of `perf::Options` since they were kind of tightly coupled. When the `event_filter` option is `None`, we have to set it to `Some()` as we're passing over the data, which also effects `EventFilterState`. I thought of it more as "state" that changes as we pass over the data as opposed to an "upfront configuration option," even thought it's that as well. In any case, I'll think about doing this differently so we end up with a clean `opt` field on `perf::Folder`, but that was the original reason why I broke things out. The "state" vs. "configuration" distinction is helpful to keep for people coming along later.